model/src/main/kotlin/config/PluginConfiguration.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2023 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.config
+
+import com.fasterxml.jackson.annotation.JsonIgnore
+
+import org.ossreviewtoolkit.utils.common.ConfigurablePluginFactory
+import org.ossreviewtoolkit.utils.common.Options
+
+/**
+ * The configuration of [configurable plugins][ConfigurablePluginFactory].
+ */
+data class PluginConfiguration(
```

Comment:
Is it really worth to introduce this class instead of just reusing `ProviderPluginConfiguration`? That's at least what I did in my [draft](https://github.com/oss-review-toolkit/ort/pull/6613/commits/5cfca0165879caeb7dc1cca111541010149f52a8) by renaming `PackageCurationProviderConfiguration` (meanwhile called `ProviderPluginConfiguration`) to `PluginConfiguration`.

Reply:
I don't like reusing `ProviderPluginConfiguration` because it has the properties `type`, `id`, and `enabled` which are all meaningless here but would still be available via the API.

Reply:
> it has the properties `type`, `id`, and `enabled` which are all meaningless here

I'm not yet convinced. `type` actually is the name in this context, and of course also scanners have names. `enabled` could become useful for scanners, too, e.g. when introducing a new experimental scanner for wider testing. Only `id` is not useful here currently, but that seems acceptable in exchange for more simplicity in the class hierarchy.

Reply:
`type` is used for package curation and configuration providers because their configuration is a list and there can be multiple instance of the same type, for scanners that is not the case. I see the issues illustrated below with the `type` property:

```yaml
# Current map
ScanCode:
  options:
    ...
  secrets:
    ...

# Using list approach like for provider plugins:
- type: ScanCode
  options:
    ...
  secrets:
    ..
- type: ScanCode # Second ScanCode entry is valid configuration syntax that needs additional checks at runtime.
  options:
    ...
  secrets:
    ..

# Using `ProviderPluginConfiguration` with the current map approach is no option because it would make this valid configuration:
ScanCode:
  type: ScanCode
  options:
    ...
FossId:
  type: ScanCode # Valid syntax but invalid configuration that needs additional checks at runtime.
```

Scanners are currently enabled via the `--scanners` and `--project-scanners` command line options. The `enabled` flag cannot be mapped to that concept, so what should it actually mean if the flag is set to `true` or `false`?

Reply:
> `type` is used for package curation and configuration providers because their configuration is a list and there can be multiple instance of the same type

I guess you mixed up `type` and `id`?

> Scanners are currently enabled via the `--scanners` and `--project-scanners` command line options. The `enabled` flag cannot be mapped to that concept, so what should it actually mean if the flag is set to `true` or `false`?

An, looks like now I mixed up `Plugin.isEnabledByDefault` and `ProviderPluginConfiguration.enabled`.

Reply:
> > `type` is used for package curation and configuration providers because their configuration is a list and there can be multiple instance of the same type
> 
> I guess you mixed up `type` and `id`?

What I meant is that in the map use for scanner configurations the type is the string key of the map (`Map<String, PluginConfiguration>`), so there can be only one instance per type by design. For the providers the map does not work and it was required to move the type into the configuration class because there multiple configurations for the same type should be possible.
That means, if we use a configuration class that contains the type we run into the issues illustrated above as for scanners (and advisors) there can be only one instance for each plugin type.